src/game_api/rpc.cpp

```diff
@@ -1354,6 +1355,33 @@ void add_entity_to_liquid_collision(uint32_t uid, bool add)
     }
 }
 
+uint8_t get_liquids_at(ENTITY_MASK liquid_mask, float x, float y, LAYER layer)
+{
+    uint8_t actual_layer = enum_to_layer(layer);
+    LiquidPhysics* liquid_physics = State::get().ptr()->liquid_physics;
+    if (actual_layer != get_liquid_layer())
+        return 0;
+    if (y >= 125.5 || y < 0 || x >= 85.5 || x < 0) // Game uses `>` instead of `>=`, resulting in it being able to read out of bounds of the array, but it never happens without OL because of the border blocks
+        return 0;
+
+    uint32_t ix = static_cast<int>((x + 0.5) / 0.3333333);
+    uint32_t iy = static_cast<int>((y + 0.5) / 0.3333333);

```

Comment:
I don't think it changes much with adding the `=` mark, since it's hard for floats to be exact equal, should probably first convert to int, maybe also then you could use the `g_level_max_x` and `g_level_max_y` values (from layer.hpp)

Also feel like on every case we convert float to int slightly different

Reply:
oh, like checking array bounds after converting to int? makes sense

Reply:
should be solved, idk if I do mark this as solved or the reviewer is the one that should do it

Reply:
Don't think it matters. As PR creator i think you always have the ability to "resolve", so it would kind of suggest that the creator should do it?. 

Anyway, I think the rule should be: either reviewer merges the PR or allows for it to be merged. And in both cases it's up to the reviewer to make sure all the issues have been resolved

Reply:
Don't think it matters. As PR creator i think you always have the ability to "resolve", so it would kind of suggest that the creator should do it?. 

Anyway, I think the rule should be: either reviewer merges the PR or allows for it to be merged. And in both cases it's up to the reviewer to make sure all the issues have been resolved